### PR TITLE
resources: add user requests endpoint

### PR DESCRIPTION
**Pull request by zzacharo**
:heart: Thank you for your contribution!

### Description

Please describe briefly your pull request.

### Checklist

Ticks in all boxes and 🟢 on all GitHub actions status checks are required to merge:

- [ ] I'm aware of the [code of conduct](https://inveniordm.docs.cern.ch/contribute/code-of-conduct/).
- [ ] I've created [logical separate commits](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commits) and followed the [commit message format](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commit-message).
- [ ] I've added relevant test cases.
- [ ] I've added relevant documentation.
- [ ] I've marked [translation strings](https://inveniordm.docs.cern.ch/develop/best-practices/i18n/) (for relevant code).
- [ ] I've followed the [CSS/JS](https://inveniordm.docs.cern.ch/develop/best-practices/css-js/) and [React](https://inveniordm.docs.cern.ch/develop/best-practices/react/) guidelines (for relevant code).
- [ ] I've followed the [web accessibility](https://inveniordm.docs.cern.ch/develop/best-practices/accessibility/) guidelines (for relevant code).
- [ ] I've followed the [user interface](https://inveniordm.docs.cern.ch/develop/best-practices/ui/) guidelines (for relevant code).
- [ ] I've identified the [copyright holder(s)](https://inveniordm.docs.cern.ch/contribute/copyright-policy/) and updated copyright headers for touched files (>15 lines contributions).
- [ ] I've NOT included third-party code (copy/pasted source code or new dependencies).

**Third-party code**

If you've added [third-party code (copy/pasted or new dependencies)](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#third-party-codedependencies), please reach out to an [architect](https://github.com/orgs/inveniosoftware/teams/architects/members).

**Reminder**

By using GitHub, you have already agreed to the [GitHub’s Terms of Service](https://help.github.com/articles/github-terms-of-service/#6-contributions-under-repository-license) including that:

1. You license your contribution under the same terms as the current repository’s license.
2. You agree that you have the right to license your contribution under the current repository’s license.

### CODE REVIEW

File: invenio_requests/services/permissions.py [hunk 1]
```diff
@@ -29,6 +29,8 @@ class PermissionPolicy(RecordPermissionPolicy):
     # Just about ability to perform a search (not what requests you can access)
     can_search = [AuthenticatedUser(), SystemProcess()]
 
+    can_search_user_requests = [AuthenticatedUser(), SystemProcess()]
```
Comment (zzacharo) [inline invenio_requests/services/permissions.py:32]: Is that correct?

File: invenio_requests/resources/requests/resource.py [hunk 2]
```diff
@@ -29,15 +29,34 @@
 class RequestsResource(RecordResource):
     """Resource for generic requests."""
 
+    def create_blueprint(self, **options):
```
Comment (ppanero) [inline invenio_requests/resources/requests/resource.py:32]: I'm just curious about this, where is it being called/installed?

**Review by ppanero — APPROVED** [review]

File: invenio_requests/resources/requests/resource.py (hunk 2 above)
Comment (zzacharo) [reply invenio_requests/resources/requests/resource.py:32]: This is used in [as_blueprint](https://github.com/inveniosoftware/flask-resources/blob/master/flask_resources/resources.py#L150) method when we register the resources as blueprints.

Comment (ppanero) [reply invenio_requests/resources/requests/resource.py:32]: thanks

File: invenio_requests/resources/requests/resource.py [hunk 3]
```diff
@@ -54,6 +73,23 @@ def search(self):
...
+    def search_user_requests(self):
+        """Perform a search over user requets.
+
+        /GET /user/requests
+        """
+        hits = self.service.search_user_requests(
+            identity=g.identity,
+            params=resource_requestctx.args,
+            es_preference=es_preference(),
+            expand=resource_requestctx.args["expand"],
```
Comment (ntarocco) [inline invenio_requests/resources/requests/resource.py:89]: Can you please change this to:
```suggestion
            expand=resource_requestctx.args.get("expand", False),
```

File: invenio_requests/services/requests/service.py [hunk 4]
```diff
@@ -255,3 +257,37 @@ def execute_action(self, identity, id_, action, data=None, uow=None,
...
+        ).execute()
+
+        return self.result_list(
+            self,
+            identity,
+            search_result,
+            params,
+            links_tpl=LinksTemplate(
+                self.config.links_user_requests_search, context={"args": params}),
+            links_item_tpl=self.links_item_tpl,
```
Comment (ntarocco) [inline invenio_requests/services/requests/service.py:292]:
```suggestion
            links_item_tpl=self.links_item_tpl,
            expandable_fields=self.expandable_fields,
            expand=expand,
```

File: invenio_requests/services/requests/service.py [hunk 5]
```diff
@@ -255,3 +257,37 @@ def execute_action(self, identity, id_, action, data=None, uow=None,
             expandable_fields=self.expandable_fields,
             expand=expand,
         )
+
+    def search_user_requests(
+            self, identity, params=None, es_preference=None, **kwargs):
```
Comment (ntarocco) [inline invenio_requests/services/requests/service.py:262]:
```suggestion
            self, identity, params=None, es_preference=None, expand=False, **kwargs):
```

**Review by ntarocco — APPROVED** [review]

**Pull request merged**